git_remote_s3/remote.py

```diff
@@ -146,6 +146,17 @@ def cmd_push(self, args: str) -> str:
                 if not force_push and not git.is_ancestor(remote_sha, sha):
                     return f'error {remote_ref} "remote ref is not ancestor of {local_ref}."?\n'
 
+            # Create and push a zip archive next to the bundle file
```

Comment:
I would move this operation to after we remove the previous bundle to minimize the likeliness that two concurrent push might occur as this is dependent on the delay between git reading the current remove commit SHA (from the bundle) and that bundle to be removed

Reply:
Good catch, I moved it after:
```
            if remote_to_remove:
                self.s3.delete_object(Bucket=self.bucket, Key=remote_to_remove)
```
in the last commit.